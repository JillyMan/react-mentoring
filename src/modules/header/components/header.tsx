import React from 'react';
import SearchBox from './search';
import { Box, Grid, Paper } from '@mui/material';
import { MainLogo } from 'modules/shared/components/main-logo';
import { ConfigrationMovieAddButtonContainer } from 'modules/configuration-movie/containers/configuration-movie-add-button-container';
import BackgroundImage from 'assets/imges/header-bg.png';

interface Props {
    searchValue: string;
    onSearchClick: (search: string) => void;
}

const styles = {
    paperContainer: {
        backgroundImage: `url(${BackgroundImage})`,
        opacity: 1.0,
        backgroundPosition: 'center center',
        backgroundSize: 'cover',
    },
    searchTitle: {
        fontFamily: 'Montserrat',
        fontWeight: 300,
        fontSize: '40px',
        lineHeight: '49px',
    },
};

export const Header = ({ searchValue, onSearchClick }: Props) => {
    return (
        <Paper style={styles.paperContainer}>
            <Box sx={{ width: '100%' }}>
                <Grid container columnSpacing={{ xs: 1, sm: 2, md: 3 }}>
                    <Grid item xs={1} />
                    <Grid item xs={8}>
                        <MainLogo />
                    </Grid>
                    <Grid item xs={2}>
                        <ConfigrationMovieAddButtonContainer />
                    </Grid>
                    <Grid item xs={1} />
                    <Grid item xs={12} sx={{ marginLeft: '150px' }}>
                        <p style={styles.searchTitle}>FIND YOUR MOVIE</p>
                    </Grid>
                    <Grid
                        item
                        xs={12}
                        sx={{ marginLeft: '150px', marginBottom: '100px' }}
                    >
                        <SearchBox
                            searchValue={searchValue}
                            text='What do your want to watch?'
                            onSearchClick={(s: string) => onSearchClick(s)}
                        />
                    </Grid>
                </Grid>
            </Box>
        </Paper>
    );
};
