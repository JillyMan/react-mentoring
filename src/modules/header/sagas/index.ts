import { addMovieSaga } from './header-saga';

export const headerSagas = [addMovieSaga];
