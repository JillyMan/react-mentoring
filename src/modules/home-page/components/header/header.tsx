import React, { useState } from 'react';
import Box from '@mui/material/Box';
import Grid from '@mui/material/Grid';
import SearchBox from './search';
import AddIcon from '@mui/icons-material/Add';
import Button from '@mui/material/Button';
import BackgroundImage from '../../../../assets/imges/header-bg.png';
import Paper from '@mui/material/Paper';
import { MainLogo } from '../shared/main-logo';

interface Props {
    onAddMovieClick: () => void;
    onSearchClick: (search: string) => void;
}

const styles = {
    paperContainer: {
        backgroundImage: `url(${BackgroundImage})`,
        opacity: 1.0,
        backgroundPosition: 'center center',
        backgroundSize: 'cover',
    },
    searchTitle: {
        fontFamily: 'Montserrat',
        fontWeight: 300,
        fontSize: '40px',
        lineHeight: '49px',
    },
    searchButton: {
        width: 150,
        height: 46,
        marginTop: '22px',
        color: '#F65261',
        borderColor: '#F65261',
        ':hover': {
            borderColor: '#F65261',
        },
    },
};

export const Header = ({ onAddMovieClick, onSearchClick }: Props) => {
    return (
        <Paper style={styles.paperContainer} square>
            <Box sx={{ width: '100%' }}>
                <Grid container columnSpacing={{ xs: 1, sm: 2, md: 3 }}>
                    <Grid item xs={1} />
                    <Grid item xs={8}>
                        <MainLogo />
                    </Grid>
                    <Grid item xs={2}>
                        <Button
                            variant='outlined'
                            startIcon={<AddIcon />}
                            sx={styles.searchButton}
                            onClick={() => onAddMovieClick()}
                        >
                            {' '}
                            Add Movie
                        </Button>
                    </Grid>
                    <Grid item xs={1} />
                    <Grid item xs={12} sx={{ marginLeft: '150px' }}>
                        <p style={styles.searchTitle}>FIND YOUR MOVIE</p>
                    </Grid>
                    <Grid
                        item
                        xs={12}
                        sx={{ marginLeft: '150px', marginBottom: '100px' }}
                    >
                        <SearchBox
                            text='What do your want to watch?'
                            onSearchClick={(s: string) => onSearchClick(s)}
                        />
                    </Grid>
                </Grid>
            </Box>
        </Paper>
    );
};
