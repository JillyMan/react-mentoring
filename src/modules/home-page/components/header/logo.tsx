import React from 'react';

interface Props {
    text: string;
    boldText: string;
}

export const Logo = ({ boldText, text }: Props) => {
    return (
        <p style={{
            width: '151px',
            height: '24px',
            color: '#F65261',
            fontFamily: 'Montserrat',
            fontSize: '20px'
        }}><b>{boldText}</b>{text}</p>
    );
}
