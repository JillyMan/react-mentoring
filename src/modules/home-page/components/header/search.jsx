import React, { useState } from 'react';
import PropTypes from 'prop-types';
import TextField from '@mui/material/TextField';
import Grid from '@mui/material/Grid';
import Button from '@mui/material/Button';
import { styled } from '@mui/material/styles';

const HeightItem = 57;
const BackgroundColor = '#F65261';

const styles = {
    searchInputStyle: {
        width: '100%',
        height: HeightItem,
    },
    searchButton: {
        width: '40%',
        height: HeightItem,
        background: BackgroundColor,
        ':hover': { background: BackgroundColor },
    },
};

const CssTextField = styled(TextField)({
    '& label.Mui-focused': {
        color: 'rgba(50, 50, 50, 0.8)',
    },
    '& .MuiOutlinedInput-root': {
        '&.Mui-focused fieldset': {
            borderColor: 'rgba(50, 50, 50, 0.8)',
        },
    },
});

class SearchBox extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            searchText: '',
        };

        this.onSearchChange = this.onSearchChange.bind(this);
    }

    onSearchChange(value) {
        this.setState({ searchText: value });
    }

    render() {
        return (
            <Grid container spacing={2}>
                <Grid item xs={8}>
                    <CssTextField
                        id='outlined-search'
                        label={this.props.text}
                        type='search'
                        onChange={(e) => this.onSearchChange(e.target.value)}
                        sx={styles.searchInputStyle}
                    />
                </Grid>
                <Grid item xs={4}>
                    <Button
                        variant='contained'
                        onClick={() => this.props.onSearchClick(this.state.searchText)}
                        sx={styles.searchButton}
                    >
                        SEARCH
                    </Button>
                </Grid>
            </Grid>
        );
    }
}

SearchBox.propTypes = {
    text: PropTypes.string,
    onSearchClick: PropTypes.func,
};

export default SearchBox;
