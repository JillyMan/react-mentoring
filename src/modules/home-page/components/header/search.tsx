import React, { useState } from 'react';
import TextField from '@mui/material/TextField';
import Grid from '@mui/material/Grid';
import Button from '@mui/material/Button';

const HeightItem = 57;
const BackgroundColor = '#F65261';

interface Props {
    text: string;
    onSearchClick: (search: string) => void;
}

export const SearchBox = ({ text, onSearchClick }: Props) => {
    const [seachText, setSearchText] = useState('');

    return (
        <Grid container spacing={2}>
            <Grid item xs={8}>
                <TextField
                    id="outlined-search"
                    label={text}
                    type="search"
                    onChange={(e) => setSearchText(e.target.value)}
                    sx={{
                        width: '100%',
                        height: HeightItem,
                        background: 'rgba(50, 50, 50, 0.8)',
                        // todo: doesn't work, don't forget to fix it!!
                        ':focus': {
                            borderColor: BackgroundColor,
                            outlineColor: 'white'
                        },
                    }}
                />
            </Grid>
            <Grid item xs={4}>
                <Button
                    variant="contained"
                    onClick={() => onSearchClick(seachText)}
                    sx={{
                        width: '40%',
                        height: HeightItem,
                        background: BackgroundColor,
                        ':hover': { background: BackgroundColor },
                    }}
                >
                    SEARCH
                </Button>
            </Grid>
        </Grid>
    );
};
