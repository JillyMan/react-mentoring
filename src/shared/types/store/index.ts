export * from './action';
export * from './app-state';
