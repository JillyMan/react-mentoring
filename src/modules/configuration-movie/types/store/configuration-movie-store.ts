import { MovieConfig } from 'shared/types/movies';

export interface ConfigurationMovieStore {
    draftConfig: MovieConfig;
    changed: boolean;
}
