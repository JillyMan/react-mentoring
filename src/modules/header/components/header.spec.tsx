import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { Header } from './header';
import { Provider } from 'react-redux';
import { configureStore } from 'store/store';

const store = configureStore();

const renderInContext = () => {
    return render(
        <Provider store={store}>
            <Header onSearchClick={() => {}} searchValue='Batman' />
        </Provider>,
    );
};

describe('Header', () => {
    test('render header with search value', () => {
        renderInContext();
        expect(screen.getByDisplayValue('Batman')).not.toBeNull();
    });

    test('render header snapsot test', () => {
        const { asFragment } = renderInContext();
        expect(asFragment()).toMatchSnapshot();
    });

    test('fire onSearchClick event', () => {
        renderInContext();
        const searchText = 'Elden ring';
        const element = screen.getByTestId('search-input');
        fireEvent.change(element, {
            target: { value: searchText },
        });
        expect(element.value).toBe(searchText);
    });
});
