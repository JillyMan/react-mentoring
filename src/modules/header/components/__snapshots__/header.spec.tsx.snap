// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render header snapsot test 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
    style="background-image: url(test-file-stub); opacity: 1; background-position: center center; background-size: cover;"
  >
    <div
      class="MuiBox-root css-8atqhb"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined css-cpk47q-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1fdpgu7-MuiGrid-root"
        />
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-12dpnmv-MuiGrid-root"
        >
          <p
            style="width: 151px; height: 24px; color: rgb(246, 82, 97); font-family: Montserrat; font-size: 20px;"
          >
            <b>
              netflix
            </b>
            roulette
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
        >
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root css-2gb2hc-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
              </svg>
            </span>
            Add Movie
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-1fdpgu7-MuiGrid-root"
        />
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-fwp9kw-MuiGrid-root"
        >
          <p
            style="font-family: Montserrat; font-weight: 300; font-size: 40px; line-height: 49px;"
          >
            FIND YOUR MOVIE
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1m4qs9w-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-12dpnmv-MuiGrid-root"
            >
              <div
                class="MuiFormControl-root MuiTextField-root ForwardRef(TextField)-root-4 css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for="outlined-search"
                  id="outlined-search-label"
                >
                  What do your want to watch?
                </label>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputTypeSearch css-1y7nk27-MuiInputBase-input-MuiOutlinedInput-input"
                    data-testid="search-input"
                    id="outlined-search"
                    type="search"
                    value="Batman"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-186xcr5"
                    >
                      <span>
                        What do your want to watch?
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1744jwa-MuiGrid-root"
            >
              <button
                class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root ForwardRef(Button)-root-5 ForwardRef(Button)-root-6 css-17nvjuj-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                SEARCH
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
